### PR TITLE
Add release notes for 19.2.0-alpha.20190701

### LINKED ISSUE #4719: v19.2.0-alpha.20190701 release notes (7/1)
Cockroach issue: https://github.com/cockroachdb/cockroach/issues/38381
https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/48431105/Release+Schedule

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes #4719.

### CODE REVIEW

File: _config_base.yml [hunk 1]
```diff
@@ -47,10 +47,10 @@ release_info:
     start_time: 2018-06-17 15:10:52.34274101 +0000 UTC
   v19.2:
     name: v19.2.0
-    version: v19.2.0-alpha.20190606
+    version: v19.2.0-alpha.20190701
```
Comment (Amruta-Ranade) [inline _config_base.yml]: Just ICYMI: Date needs to be updated throughout. Apologies if it was already on your to-do list.

File: releases/v19.2.0-alpha.20190701.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+title: What&#39;s New in v19.2.0-alpha.20190701
+toc: true
+summary: Additions and changes in CockroachDB version v19.2.0-alpha.20190701 since version v19.2.0-alpha.20190606
+---
+
+## July 01, 2019
+
+This release contains enhancements and bug fixes to the 19.2.0 series, including:
+
+- An initial prototype of support for using `IMPORT INTO` to ingest CSV data into an existing table. Usage: `IMPORT INTO t(a, b) CSV DATA ('gs://acme-co/employees.csv')`.
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Might wanna change "An initial prototype of support" > "Added experimental support"

File: releases/v19.2.0-alpha.20190701.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+### Enterprise edition changes
+
+- Fixes help text that erroneously labeled [Encryption at Rest](../v19.2/encryption.html) as experimental. [#38228][#38228]
+
+### SQL language changes
+
+- An initial prototype of support for using `IMPORT INTO` to ingest CSV data into an existing table. Usage: `IMPORT INTO t(a, b) CSV DATA ('gs://acme-co/employees.csv')`. [#37451][#37451]
+- Implements sql support for timestamp objects to have an optional precision value. [#37920][#37920]
+- The logging format for statement execution (as enabled by `sql.trace.log_statement_execute`) and audit logs (as enabled by [`ALTER TABLE .. EXPERIMENTAL AUDIT SET ...`](../v19.2/experimental-audit.html)) is extended with one column that reports internal execution retries performed automatically by the server. The value zero is logged for the first retry attempt and thus corresponds to statements directly received from a client. [#38028][#38028]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: nit: The value zero > The value `zero`

File: releases/v19.2.0-alpha.20190701.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+- Add `WITH EXPIRATION` option for [`SPLIT AT`](../v19.2/split-at.html). [#38004][#38004]
+- The `DISTINCT` keyword is now rejected when used in a window function. It was previously accepted, but ignored. [#37912][#37912]
+- Add `UNSPLIT ALL` statement. [#38023][#38023]
+- The `relam` column of the `pg_catalog` tables `pg_class` and `pg_index` is now filled in correctly. [#38164][#38164]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Just checking: should `relam` be `realm`?

File: releases/v19.2.0-alpha.20190701.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+- CockroachDB now supports the special case `set timezone = 'utc'` as a special alias for `set timezone = 'UTC'`. The other time zone names are still case-sensitive as previously, pending resolution of issue #36864. [#38195][#38195]
+- Add support for `ALTER TABLE ... ALTER COLUMN ... SET NOT NULL`, which adds a non-null constraint to an existing column. [#37554][#37554]
+- `pg_get_constraintdef` now outputs `CHECK` constraints exactly the same way that Postgres does. [#38318][#38318]
+- The `format_type` Postgres builtin now properly respects its second argument. [#38318][#38318]
+- [`SHOW CREATE TABLE`](../v19.2/show-create.html) now prints foreign keys more like Postgres does. [#38318][#38318]
+- The `pg_type_is_visible` builtin from Postgres is now implemented. [#38318][#38318]
+- Casting an int2vector to a string now produces a Postgres-compatible result. [#38318][#38318]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: int2vector > `int2vector`

File: releases/v19.2.0-alpha.20190701.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+### Command-line changes
+
+- CockroachDB will now print out an error message and an informative hint if the time zone database is unusable. [#38197][#38197]
+
+### Admin UI changes
+
+- The snapshots graph on the [Replication page](../v19.2/admin-ui-replication-dashboard.html) in the Web UI now includes learner snapshots. [#38147][#38147]
+- Remove version information from login screen. [#38140][#38140]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Remove > Removed

File: releases/v19.2.0-alpha.20190701.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+- Adding `Implicit Transaction` flag to [Statements](../v19.2/admin-ui-statements-page.html) and [Statement Details](../v19.2/admin-ui-statements-page.html#statement-details-page) page. [#36341][#36341]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Adding > Added

File: releases/v19.2.0-alpha.20190701.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+- Add `WITH EXPIRATION` option for [`SPLIT AT`](../v19.2/split-at.html). [#38004][#38004]
+- The `DISTINCT` keyword is now rejected when used in a window function. It was previously accepted, but ignored. [#37912][#37912]
+- Add `UNSPLIT ALL` statement. [#38023][#38023]
+- The `relam` column of the `pg_catalog` tables `pg_class` and `pg_index` is now filled in correctly. [#38164][#38164]
+- CockroachDB now supports the special case `set timezone = 'utc'` as a special alias for `set timezone = 'UTC'`. The other time zone names are still case-sensitive as previously, pending resolution of issue #36864. [#38195][#38195]
+- Add support for `ALTER TABLE ... ALTER COLUMN ... SET NOT NULL`, which adds a non-null constraint to an existing column. [#37554][#37554]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Add > Added

File: releases/v19.2.0-alpha.20190701.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+### Enterprise edition changes
+
+- Fixes help text that erroneously labeled [Encryption at Rest](../v19.2/encryption.html) as experimental. [#38228][#38228]
+
+### SQL language changes
+
+- An initial prototype of support for using `IMPORT INTO` to ingest CSV data into an existing table. Usage: `IMPORT INTO t(a, b) CSV DATA ('gs://acme-co/employees.csv')`. [#37451][#37451]
+- Implements sql support for timestamp objects to have an optional precision value. [#37920][#37920]
+- The logging format for statement execution (as enabled by `sql.trace.log_statement_execute`) and audit logs (as enabled by [`ALTER TABLE .. EXPERIMENTAL AUDIT SET ...`](../v19.2/experimental-audit.html)) is extended with one column that reports internal execution retries performed automatically by the server. The value zero is logged for the first retry attempt and thus corresponds to statements directly received from a client. [#38028][#38028]
+- Add `WITH EXPIRATION` option for [`SPLIT AT`](../v19.2/split-at.html). [#38004][#38004]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Add > Added

File: releases/v19.2.0-alpha.20190701.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+- The `DISTINCT` keyword is now rejected when used in a window function. It was previously accepted, but ignored. [#37912][#37912]
+- Add `UNSPLIT ALL` statement. [#38023][#38023]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Add > Added

File: releases/v19.2.0-alpha.20190701.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+- Upgrade the build environment to use go1.12.5. [#37966][#37966]
+
+### Enterprise edition changes
+
+- Fixes help text that erroneously labeled [Encryption at Rest](../v19.2/encryption.html) as experimental. [#38228][#38228]
+
+### SQL language changes
+
+- An initial prototype of support for using `IMPORT INTO` to ingest CSV data into an existing table. Usage: `IMPORT INTO t(a, b) CSV DATA ('gs://acme-co/employees.csv')`. [#37451][#37451]
+- Implements sql support for timestamp objects to have an optional precision value. [#37920][#37920]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Implements > Implemented

File: releases/v19.2.0-alpha.20190701.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+### Admin UI changes
+
+- The snapshots graph on the [Replication page](../v19.2/admin-ui-replication-dashboard.html) in the Web UI now includes learner snapshots. [#38147][#38147]
+- Remove version information from login screen. [#38140][#38140]
+- Adding `Implicit Transaction` flag to [Statements](../v19.2/admin-ui-statements-page.html) and [Statement Details](../v19.2/admin-ui-statements-page.html#statement-details-page) page. [#36341][#36341]
+
+### Bug fixes
+
+- The `max_retries` column in the statement statistics now properly displays the maximum number of times a statement was retried. [#38028][#38028]
+- Previously, due to a bug when estimating result set sizes in the optimizer, queries involving int ranges that were very large could result in poor plans being generated. [#38036][#38036]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: int > `int`

File: releases/v19.2.0-alpha.20190701.md [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+- Fix bug where MVCC value at future timestamp is returned after a transaction restart. [#38085][#38085]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Fix bug > Fixed a bug

File: releases/v19.2.0-alpha.20190701.md [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+- The YCSB workload will no longer ignore the `--db` option. [#38236][#38236]
+- Localities in the `crdb_internal` tables were displayed JSON objects instead of strings. This format stripped them of their order, which is important in localities. This string format now exactly matches what is used in the command line for setting localities. [#37944][#37944]
+- Fixed an incorrect type mismatch error when empty array values are used as default values (and potentially in other contexts). [#38294][#38294]
+- Fix a panic that could occur when decoding decimals as query parameters. [#38304][#38304]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Fix > Fixed

File: releases/v19.2.0-alpha.20190701.md [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+- Nulls are now correctly handled by `MIN`, `SUM`, and `AVG` [aggregate functions](../v19.2/functions-and-operators.html#aggregate-functions) when used as [window functions](../v19.2/window-functions.html). [#38110][#38110]
+- Prevent a possible panic during [`JSONB`](../v19.2/jsonb.html) binary query parameter parsing. [#38346][#38346]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: Prevent > Prevented 
(Need to make the change in the next three issues as well)

File: releases/v19.2.0-alpha.20190701.md [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+- Prevent a possible panic during [`INET`](../v19.2/inet.html) binary query parameter parsing. [#38346][#38346]
+- Prevent a possible panic during [`VARBIT`](../v19.2/bit.html) binary query parameter parsing. [#38346][#38346]
+- Prevent a possible panic during array binary query parameter parsing. [#38346][#38346]
+- Fixed possible deadlock when storage engine write fails. [#38170][#38170]
+- Fixed potential reappearance of deleted timeseries data, which could trip consistency checker. [#38170][#38170]
+- Removed dependency on `sync_file_range` on Linux platforms on which it returns `ENOSYS`, such as WSL (Windows Subsystem for Linux). [#38170][#38170]
+
+### Build changes
+
+- A recent libedit is now bundled with CockroachDB, which fixes some line editing bugs in the CockroachDB console. On platforms that include libedit as part of the base system, like macOS and FreeBSD, CockroachDB no longer links against the system libedit. [#32623][#32623]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: libedit > `libedit`

File: releases/v19.2.0-alpha.20190701.md [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+
+### Miscellaneous
+
+#### Advanced debugging
+
+- Checking the "skip should queue" checkbox in the Manual Enqueue Range [advanced debugging page](../v19.2/admin-ui-debug-pages.html) now works for the GC Queue. [#38249][#38249]
+
+#### Security
+
+- Only check CN on first certificate in file. [#38163][#38163]
```
Comment (Amruta-Ranade) [inline releases/v19.2.0-alpha.20190701.md]: CN > `CN`

Comment (rmloveland) [reply releases/v19.2.0-alpha.20190701.md]: Fixed.

File: releases/v19.2.0-alpha.20190701.md (hunk 16 above)
Comment (rmloveland) [reply releases/v19.2.0-alpha.20190701.md]: Fixed.

File: releases/v19.2.0-alpha.20190701.md (hunk 15 above)
Comment (rmloveland) [reply releases/v19.2.0-alpha.20190701.md]: Fixed.

File: releases/v19.2.0-alpha.20190701.md (hunk 14 above)
Comment (rmloveland) [reply releases/v19.2.0-alpha.20190701.md]: Fixed.

File: releases/v19.2.0-alpha.20190701.md (hunk 13 above)
Comment (rmloveland) [reply releases/v19.2.0-alpha.20190701.md]: Fixed.

File: releases/v19.2.0-alpha.20190701.md (hunk 5 above)
Comment (rmloveland) [reply releases/v19.2.0-alpha.20190701.md]: Fixed.

File: releases/v19.2.0-alpha.20190701.md (hunk 4 above)
Comment (rmloveland) [reply releases/v19.2.0-alpha.20190701.md]: Great question! I double checked and it's a thing: https://www.postgresql.org/docs/9.6/catalog-pg-class.html

File: releases/v19.2.0-alpha.20190701.md (hunk 3 above)
Comment (rmloveland) [reply releases/v19.2.0-alpha.20190701.md]: Fixed.

File: releases/v19.2.0-alpha.20190701.md (hunk 2 above)
Comment (rmloveland) [reply releases/v19.2.0-alpha.20190701.md]: Fixed.

File: _config_base.yml (hunk 1 above)
Comment (rmloveland) [reply _config_base.yml]: Just confirmed, turns out we're sticking with `20190701`

**Pull request merged**